app/src/main/java/org/wikipedia/page/PageFragmentLoadState.kt

```diff
@@ -149,18 +150,29 @@ class PageFragmentLoadState(private var model: PageViewModel,
                     }
                     val makeWatchRequest = WikipediaApp.instance.isOnline && AccountUtil.isLoggedIn
                     val watchedRequest = async {
-                        if (makeWatchRequest) {
-                            ServiceFactory.get(title.wikiSite).getWatchedStatusWithCategories(title.prefixedText)
-                        } else if (WikipediaApp.instance.isOnline && !AccountUtil.isLoggedIn) {
-                            AnonymousNotificationHelper.observableForAnonUserInfo(title.wikiSite)
-                        } else {
+                        try {
+                            if (makeWatchRequest) {
+                                ServiceFactory.get(title.wikiSite)
+                                    .getWatchedStatusWithCategories(title.prefixedText)
+                            } else if (WikipediaApp.instance.isOnline && !AccountUtil.isLoggedIn) {
+                                AnonymousNotificationHelper.maybeGetAnonUserInfo(title.wikiSite)
+                            } else {
+                                MwQueryResponse()
+                            }
+                        } catch (_: IOException) {
```

Comment:
Very importantly, note that I'm only catching `IOException`, which will limit the catching to network-related issues. Other errors, such as coroutine termination or Mediawiki errors, will continue to have the same effect as before.